File: aoc-2015/src/test/java/com/capital7software/aoc/aoc2015aoc/days/Day15Test.java
package com.capital7software.aoc.aoc2015aoc.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day15Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day15Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day15();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void whatIsTheTotalScoreOfTheHighestScoringCookie() {
    var instance = new Day15();
    long expected = 62_842_880;

    var actual = instance.whatIsTheTotalScoreOfTheHighestScoringCookie(lines, 100_000);
    assertEquals(expected, actual, "The total score of the highest-scoring "
        + "cookie is not what was expected: " + expected);
  }

  @Test
  void whatIsTheTotalScoreOfTheHighestScoringCalorieRestrictedCookie() {
    var instance = new Day15();
    long expected = 57_600_000;

    var actual = instance.whatIsTheTotalScoreOfTheHighestScoringCaloriesRestrictedCookie(
        lines,
        250_000
    );
    assertEquals(expected, actual, "The total score of the highest-scoring "
        + "cookie is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2016/src/test/kotlin/com/capital7software/aoc/aoc2016aoc/days/Day15Test.kt
package com.capital7software.aoc.aoc2016aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import com.capital7software.aoc.lib.math.DiscSculpture
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day15Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day15Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day15()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testTimeToPushTheButton() {
    val instance = Day15()
    val expected = 5
    val actual = instance.timeToPushTheButton(lines)

    assertEquals(expected, actual, "The time to push the button "
        + "is not what was expected: $expected")
  }

  @Test
  fun testTimeToPushTheButtonWithAddedDisc() {
    val instance = Day15()
    val expected = 5
    val actual = instance.timeToPushTheButton(lines)

    assertEquals(expected, actual, "The time to push the button "
        + "is not what was expected: $expected")
  }

  @Test
  fun testNoSolutionToTimeToPushTheButton() {
    val instance = DiscSculpture(lines)
    instance.add(DiscSculpture.Disc(instance.size + 1, 2, 0))
    instance.add(DiscSculpture.Disc(instance.size + 1, 2, 0))
    val expected = -1
    val actual = instance.solve()

    assertEquals(expected, actual, "The time to push the button "
        + "is not what was expected: $expected")
  }
}
